CMakeLists.txt

```diff
@@ -1,42 +1,80 @@
-cmake_minimum_required(VERSION 3.5)
-project(hsm)
+cmake_minimum_required(VERSION 3.14)
 
-# Conan
-if(EXISTS ${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-    include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-    conan_basic_setup(TARGETS)
-endif()
+project(
+    hsm
+    VERSION 0.0.1
+    DESCRIPTION "Finite state machine library based on Boost.Hana"
+    HOMEPAGE_URL "https://github.com/erikzenker/hsm"
+    LANGUAGES CXX
+)
 
-# Sanatizers
-if(${HSM_ASAN})
-    set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
-    set (CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
-endif()
+# ---- Warning guard ----
 
-if(${HSM_UBSAN})
-    set(CMAKE_CXX_FLAGS "-fsanitize=undefined -fno-omit-frame-pointer -fno-sanitize-recover -fsanitize=float-divide-by-zero")
+# Protect dependents from this project's warnings if the guard isn't disabled
+set(hsm_warning_guard SYSTEM)
+if(hsm_INCLUDE_WITHOUT_SYSTEM)
+  set(hsm_warning_guard "")
 endif()
 
-if(${HSM_TIME_TRACE})
-    set(CMAKE_CXX_FLAGS "-ftime-trace")
-endif()
+# ---- Declare library ----
```

Comment:
Why was that moved from the `src` folder cmake here? I followed that template for my cmake https://github.com/erikzenker/cmake-project-template which is based on https://www.youtube.com/watch?v=S4QSKLXdTtA

Reply:
It's a personal preference. When I look at the root CML, I expect to see **all** the build and usage requirements of the library, without having to chase many CML files.